swan-lake/learn/deployment/docker.md

```diff
@@ -0,0 +1,1108 @@
+      @docker 		 - complete 2/2 
+
+      Run the following command to start a Docker container:
+      docker run -d -p 9090:9090 hello_world_docker:latest
+    ```
+
+    The artifacts files below will be generated with the build process.
+
+    ```
+    > tree
+    .
+    ├── docker
+    │   └── Dockerfile
+    ├── hello_world_docker.bal
+    └── hello_world_docker.jar
+
+    1 directory, 3 files
+
+    ```
+
+    The build process automatically generates a Dockerfile with the following content:
+
+    ```
+    # Auto Generated Dockerfile
+    FROM ballerina/jre8:v1
+
+    LABEL maintainer="dev@ballerina.io"
+
+    RUN addgroup troupe \
+        && adduser -S -s /bin/bash -g 'ballerina' -G troupe -D ballerina \
+        && apk add --update --no-cache bash \
+        && chown -R ballerina:troupe /usr/bin/java \
+        && rm -rf /var/cache/apk/*
+
+    WORKDIR /home/ballerina
+
+    COPY hello_world_docker.jar /home/ballerina
+
+    USER ballerina
+
+    CMD java -jar hello_world_docker.jar
+
+    ```
+
+
+2. Verify that the Docker image is created.
+
+    ```
+    > docker images
+    REPOSITORY          TAG         IMAGE ID            CREATED             SIZE
+    hello_world_docker  latest      e48123737a65        7 minutes ago       134MB
+
+    ```
+
+    > Since the annotation is not configured to have a custom Docker image name and tag, the build process will create a Docker image with the default values: the file name of the generated .jar file with the latest tag (e.g., `hello_world_docker:latest`).
+
+3. Run the Docker image as a container (use the below command printed in step 1).
+
+    ```
+    > docker run -d -p 9090:9090 hello_world_docker:latest
+    32461676d3c22848088390483a414e5b1d11a7a73c2296eccb18e6c9f27c41c0
+    ```
+
+
+4. Verify that the Docker container is running.
+
+    ```
+    > docker ps
+    CONTAINER ID  IMAGE    		 COMMAND    	      CREATED              STATUS            PORTS     NAMES
+    32461676d3c2  hello_world_docker:latest  "/bin/sh -c 'java -j…"   About a minute ago   Up About a minute 0.0.0.0:9090->9090/tcp    lucid_turing
+    ```
+
+
+5. Access the hello world service with the cURL command.
+
+    ```
+    > curl http://localhost:9090/hello/sayHello           
+    Hello World!
+    ```
+
+
+6. Clean up the used artifacts.
+
+    ```
+    Stop / kill running docker container
+    > docker kill 32461676d3c2
+    32461676d3c2
+
+    Remove docker container files
+    > docker em 32461676d3c2
+
+    Remove docker image
+    > docker rmi e48123737a65
+  ```
+
+### Creating a custom Ballerina Docker image and pushing it autimatically to the Docker registry
+
+This usecase shows how to run a simple Ballerina hello world service in a Docker container with annotation configurations to create a Docker image with a custom image-name and tag, and then automatically push the created image to the Docker registry.
+
+#### Prerequisites
+
+- A machine with [Docker](https://docs.docker.com/get-docker/) installed
```

Comment:
```suggestion
- A machine with [Docker](https://docs.docker.com/get-docker/) installed.
```

Reply:
We don't need a fullstop here as this is not a complete sentence.